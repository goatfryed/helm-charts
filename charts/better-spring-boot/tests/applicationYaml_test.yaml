suite: test chart
templates:
  - deployment.yaml
  - configMap.appConfig.yaml
tests:
  - it: injects application properties spring-container
    values:
      - "values/values.applicationYaml.yaml"
    asserts:
      - template: deployment.yaml
        matchSnapshot:
          path: spec.template.spec
      - template: configMap.appConfig.yaml
        matchSnapshot: {}
  - it: activates profiles automatically
    values:
      - "values/values.applicationYaml.yaml"
    templates:
      - deployment.yaml
    asserts:
      - equal:
          path: spec.template.spec.containers[0].env[0]
          value:
            name: SPRING_PROFILES_ACTIVE
            value: "profile1, profile2, profile3, profile4"
  - it: mounts configuration values
    set:
      image.repository: goatfryed/spring-boot-echo
      spring.config.userConfig.values:
        foo: bar
        how:
          are: you
    asserts:
      - template: deployment.yaml
        matchSnapshot:
          path: spec.template.spec
  - it: mounts referenced configMaps
    set:
      image.repository: goatfryed/spring-boot-echo
      spring.config.userConfig.configMap.name: user-defined-config-map
    asserts:
      - template: deployment.yaml
        matchSnapshot:
          path: spec.template.spec
  - it: mounts referenced secrets
    set:
      image.repository: goatfryed/spring-boot-echo
      spring.config.userConfig.secret.name: user-defined-secret
    asserts:
      - template: deployment.yaml
        matchSnapshot:
          path: spec.template.spec
  - it: calculates checksum over app config
    set:
      image.repository: goatfryed/spring-boot-echo
      spring.config:
          profile1:
            values:
              foo: foo
          profile2:
            values:
              bar: bar
    asserts:
      - template: deployment.yaml
        equal:
          path: spec.template.metadata.labels.release/checksum/app-config
          value: 3d9a1b9ff8249efd74f1fd196e6e4f5bf7d51fc4dcfb855844b380ce9e311c31
  - it: calculates checksum over app config with stable sort
    set:
      image:
        repository: goatfryed/spring-boot-echo
      spring:
        profiles:
          active: profile2, profile1
        config:
          profile2:
            values:
              bar: bar
          profile1:
            values:
              foo: foo
    asserts:
      - template: deployment.yaml
        equal:
          path: spec.template.metadata.labels.release/checksum/app-config
          value: 3d9a1b9ff8249efd74f1fd196e6e4f5bf7d51fc4dcfb855844b380ce9e311c31
  - it: calculates checksum over active app config
    set:
      image.repository: goatfryed/spring-boot-echo
      spring:
        profiles:
          active: profile1, profile2
        config:
          profile1:
            values:
              foo: foo
          profile2:
            values:
              bar: bar
          profile3:
            values:
              foo: bar
    asserts:
      - template: deployment.yaml
        equal:
          path: spec.template.metadata.labels.release/checksum/app-config
          value: 3d9a1b9ff8249efd74f1fd196e6e4f5bf7d51fc4dcfb855844b380ce9e311c31
