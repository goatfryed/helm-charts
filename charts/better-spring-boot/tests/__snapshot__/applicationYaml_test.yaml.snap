injects application properties spring-container:
  1: |
    containers:
      - env:
          - name: SPRING_CONFIG_ADDITIONAL_LOCATION
            value: optional:/app/config/*/
          - name: SPRING_PROFILES_ACTIVE
            value: profile1, profile2, profile3, profile4
        image: goatfryed/spring-boot-echo:latest
        imagePullPolicy: IfNotPresent
        name: spring-boot
        ports:
          - containerPort: 8080
            name: http
            protocol: TCP
          - containerPort: 8080
            name: http-management
            protocol: TCP
        resources: {}
        securityContext: {}
        volumeMounts:
          - mountPath: /app/config/helm-config
            name: helm-application-properties
            readOnly: true
          - mountPath: /app/config/config-map/application-profile3.properties
            name: user-application-properties-profile3
            readOnly: true
            subPath: my-key
          - mountPath: /app/config/config-map/application-profile4.yaml
            name: user-application-properties-profile4
            readOnly: true
            subPath: app.yml
    securityContext: {}
    serviceAccountName: RELEASE-NAME-spring-boot
    volumes:
      - configMap:
          name: RELEASE-NAME-spring-boot-app-config
        name: helm-application-properties
      - configMap:
          name: user-defined-config-map
        name: user-application-properties-profile3
      - name: user-application-properties-profile4
        secret:
          secretName: user-defined-config-map
  2: |
    apiVersion: v1
    data:
      application-profile1.yaml: |
        server:
          port: 8080
      application-profile2.yaml: |
        logging:
          level:
            org.springframework: error
            com.example.myapp: debug
    kind: ConfigMap
    metadata:
      name: RELEASE-NAME-spring-boot-app-config
mounts configuration values:
  1: |
    containers:
      - env:
          - name: SPRING_CONFIG_ADDITIONAL_LOCATION
            value: optional:/app/config/*/
          - name: SPRING_PROFILES_ACTIVE
            value: userConfig
        image: goatfryed/spring-boot-echo:latest
        imagePullPolicy: IfNotPresent
        name: spring-boot
        ports:
          - containerPort: 8080
            name: http
            protocol: TCP
          - containerPort: 8080
            name: http-management
            protocol: TCP
        resources: {}
        securityContext: {}
        volumeMounts:
          - mountPath: /app/config/helm-config
            name: helm-application-properties
            readOnly: true
    securityContext: {}
    serviceAccountName: RELEASE-NAME-spring-boot
    volumes:
      - configMap:
          name: RELEASE-NAME-spring-boot-app-config
        name: helm-application-properties
mounts referenced configMaps:
  1: |
    containers:
      - env:
          - name: SPRING_CONFIG_ADDITIONAL_LOCATION
            value: optional:/app/config/*/
          - name: SPRING_PROFILES_ACTIVE
            value: userConfig
        image: goatfryed/spring-boot-echo:latest
        imagePullPolicy: IfNotPresent
        name: spring-boot
        ports:
          - containerPort: 8080
            name: http
            protocol: TCP
          - containerPort: 8080
            name: http-management
            protocol: TCP
        resources: {}
        securityContext: {}
        volumeMounts:
          - mountPath: /app/config/helm-config
            name: helm-application-properties
            readOnly: true
          - mountPath: /app/config/config-map/application-userConfig.yaml
            name: user-application-properties-userConfig
            readOnly: true
            subPath: application.yaml
    securityContext: {}
    serviceAccountName: RELEASE-NAME-spring-boot
    volumes:
      - configMap:
          name: RELEASE-NAME-spring-boot-app-config
        name: helm-application-properties
      - configMap:
          name: user-defined-config-map
        name: user-application-properties-userConfig
mounts referenced secrets:
  1: |
    containers:
      - env:
          - name: SPRING_CONFIG_ADDITIONAL_LOCATION
            value: optional:/app/config/*/
          - name: SPRING_PROFILES_ACTIVE
            value: userConfig
        image: goatfryed/spring-boot-echo:latest
        imagePullPolicy: IfNotPresent
        name: spring-boot
        ports:
          - containerPort: 8080
            name: http
            protocol: TCP
          - containerPort: 8080
            name: http-management
            protocol: TCP
        resources: {}
        securityContext: {}
        volumeMounts:
          - mountPath: /app/config/helm-config
            name: helm-application-properties
            readOnly: true
          - mountPath: /app/config/config-map/application-userConfig.yaml
            name: user-application-properties-userConfig
            readOnly: true
            subPath: application.yaml
    securityContext: {}
    serviceAccountName: RELEASE-NAME-spring-boot
    volumes:
      - configMap:
          name: RELEASE-NAME-spring-boot-app-config
        name: helm-application-properties
      - name: user-application-properties-userConfig
        secret:
          secretName: user-defined-secret
